main.go

```diff
@@ -78,21 +80,66 @@ func main() {
 	} else if ctx.SetProtocol(gpgme.ProtocolAssuan) != nil {
 		log.Debugf("Cannot initialize Assuan IPC: %v. Disabling GPG and SSH watchers.", err)
 	} else {
-		gpgPubringPath := path.Join(gpgme.GetDirInfo("homedir"), "pubring.kbx")
-		if _, err := os.Stat(gpgPubringPath); err == nil {
-
-			requestGPGCheck := make(chan bool)
-			go detector.CheckGPGOnRequest(requestGPGCheck, notifiers, ctx)
-			go detector.WatchGPG(gpgPubringPath, requestGPGCheck)
-			go detector.WatchSSH(requestGPGCheck, exits)
-		} else {
-			log.Debugf("'%v' could not be found. Disabling GPG and SSH watchers.", gpgPubringPath)
+		var gpgPubringPath = path.Join(gpgme.GetDirInfo("homedir"), "private-keys-v1.d")
+		if _, err := os.Stat(gpgPubringPath); os.IsNotExist(err) {
+			fmt.Printf("Directory '%s' does not exist (you have no private keys).\n", gpgPubringPath)
+			return
 		}
+		var searchTerm = "shadowed-private-key"
+		var filesToWatch []string
+		filesToWatch, err := findMatchingFiles(gpgPubringPath, searchTerm)
```

Comment:
minor, but personally I'd rename the function to `findShadowedPrivateKeys`, let it only accept `gpgPrivateKeysDirPath` and it should know itself the search term to look for